src/Units/UnitContext.java

```diff
@@ -0,0 +1,105 @@
+package Units;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import CommandingOfficers.Commander;
+import Engine.GamePath;
+import Engine.XYCoord;
+import Engine.UnitMods.UnitModifier;
+import Terrain.Environment;
+import Terrain.GameMap;
+
+/**
+ * A basic struct for details relevant to calculation of unit activities.
+ * <p>Intended to be ephemeral, and thus free to modify (potentially destructively)
+ */
+public class UnitContext extends UnitState
+{
+  private static final long serialVersionUID = 1L;
+
+  // Groups are set together
+  public Unit unit;
+
+  public GamePath path;
+  public XYCoord coord;
+
+  public int maxHP;
+  public int attackPower;
+  public int defensePower;
+  public int costBase;
+  public double costMultiplier;
+  public int costShift;
+
+  public Environment env;
+  public int terrainStars = 0;
+
+  public WeaponModel weapon;
+
+  public final List<UnitModifier> mods = new ArrayList<UnitModifier>();
+
+  public UnitContext(GameMap map, Unit u, WeaponModel w, int x, int y)
+  {
+    this(u);
+    weapon = w;
+    coord = new XYCoord(x, y);
+    setEnvironment(map.getEnvironment(coord));
+  }
+  public UnitContext(Unit u)
+  {
+    super(u);
+    unit = u;
+    coord = new XYCoord(u.x, u.y);
+    heldUnits.addAll(u.heldUnits);
+    mods.addAll(u.getModifiers());
+    initModel();
+  }
+  public UnitContext(Commander co, UnitModel um)
+  {
+    super(co, um);
+    initModel();
+  }
+  public UnitContext(UnitContext other)
+  {
+    super(other);
+    unit = other.unit;
+    path = other.path;
+    coord = other.coord;
+    maxHP = other.maxHP;
+    attackPower = other.attackPower;
+    defensePower = other.defensePower;
+    env = other.env;
+    terrainStars = other.terrainStars;
+    weapon = other.weapon;
+    mods.addAll(other.mods);
+  }
```

Comment:
If possible, I always try to consolidate overloads into a single actual implementation, and have the others simply forward to the real/comprehensive one.

Reply:
There are three semantic variants of this constructor: unit, model, and copy
Model and copy have only one each, and don't really make sense to merge in that way.
I think I can make some progress on the unit ones, though.

Reply:
If you have three different use cases for this class (different enough that the constructors are irreconcilable), I'd consider whether to try and cram it all into one class.

Reply:
The copy constructor isn't a separate use case. Or at least, I sure hope it isn't.

That leaves unit and model-only cases, and they're actually used for the same things - this allows us to collect modifications and preview stuff without having a specific `Unit` in question.